### ISSUE TITLE
Signature and threading issues

### ISSUE THREAD

**Issue by vvdb** [sillsdev/icu-dotnet#54]
When getting the latest version of ICU.NET, I noticed 2 things:
1. The signature of u_charType is incorrect: the return value is not an int, but an sbyte:

```
			[UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
			internal delegate sbyte u_charTypeDelegate(int characterCode);
```
This will make the IsSymbol and GetCharType pass.

2. The initialization of the ICU4C library in multithreading scenarios is incorrect. See  [http://userguide.icu-project.org/design#TOC-ICU-Initialization-and-Termination](http://userguide.icu-project.org/design#TOC-ICU-Initialization-and-Termination). This will yield crashes in multithreaded scenarios like Lucene.Net, which are extremely puzzling.
Sadly, this isn't easily corrected: the whole NativeMethods isn't written to be thread-friendly.

**Comment by ermshiperete** [comment]
@vvdb Thanks for your feedback. I fixed the signature, and added a `Wrapper.Init()` method (#55) that should improve the situation a little bit.

Feel free to create a PR with a more complete fix.

**Comment by ermshiperete** [comment]
`Wrapper.Init()` implemented in 2.3.0-beta0008
